Applications/Spire/Source/BookView/BookViewWindow.cpp

```diff
@@ -131,7 +131,7 @@ void BookViewWindow::show_context_menu(const QPoint& pos) {
 void BookViewWindow::show_properties_dialog() {
   BookViewPropertiesDialog dialog(get_properties(), Security(), this);
   m_dialog_apply_connection = dialog.connect_apply_signal(
-    [=] (auto p) { set_properties(p); });
+    [=, &dialog] { set_properties(dialog.get_properties()); });
```

Comment:
Capture by ```&```, ie. replace ```[=, &dialog]``` with ```[&]```. Try to understand why that's the case.

Reply:
As far as I know in this case the lambda is capturing 'this' and the dialog by reference. What's the difference in capturing 'this' be reference?

Reply:
The lambda is capturing ```this``` by value and ```dialog``` by reference. My comment is that everything should be captured by reference as so to simply use ```[&]``` as the capture.

Reply:
Sorry, my comment was poorly worded. I meant 'this' and the dialog were being captured, but only the dialog was being captured by reference.

Reply:
Yep, they should both be captured by reference since the closure does not escape the lambda expression's lexical scope. By default (as-in unless you have a good reason to) if the closure does not escape the lambda's lexical scope then the capture should be by reference. If the closure does escape the lambda's lexical scope, then the capture should be by value.

Feel free to give some thought as to why this is and do your best to remember this principle.